uaclient/util.py

```diff
@@ -61,6 +62,31 @@ def encode_text(text, encoding='utf-8'):
     return text.encode(encoding)
 
 
+def get_dict_deltas(orig_dict, new_dict, path=''):
+    """Return a dictionary of delta between orig_dict and new_dict."""
+    deltas = {}
+    for key, value in orig_dict.items():
+        new_value = new_dict.get(key, DROPPED_DICT_KEY)
+        if isinstance(value, dict):
+            if path:
+                sub_path = path + '.' + key
+            else:
+                sub_path = key
+            sub_delta = get_dict_deltas(
+                value, new_dict.get(key, {}), path=sub_path)
```

Comment:
If `new_dict` doesn't have `key`, do we need to recurse to determine the outcome here? Doesn't this means it should be a `DROPPED_DICT_KEY`?

Reply:
I think this is also causing a logical error ATM:

```py
>>> from uaclient.util import get_dict_deltas

>>> get_dict_deltas({}, {})
{}

>>> get_dict_deltas({'a': {}}, {})
{}

>>> get_dict_deltas({'a': ''}, {})
Out[6]: {'a': 'DROPPED'}
```

Reply:
@blackboxsw This comment appears unaddressed (as does the one below it):

```py
In [2]: get_dict_deltas({'a': ''}, {})
Out[2]: {'a': <object at 0x7fc624a9c660>}

In [3]: get_dict_deltas({'a': {}}, {})
Out[3]: {}
```

Reply:
Just addressed dropped keys of type dict as value in commit 6bc337c8b611d2fa0ab49fd8948e8b59dcf92415